engine/src/test/java/org/camunda/bpm/engine/test/api/task/TaskQueryLastUpdatedTest.java

```diff
@@ -0,0 +1,547 @@
+    Task taskResult = taskService.createTaskQuery().processInstanceId(processInstance.getId()).taskUpdatedAfter(beforeStart).singleResult();
+    assertThat(taskResult).isNotNull();
+    assertThat(taskResult.getLastUpdated()).isNull();
+    assertThat(taskResult.getCreateTime()).isAfter(beforeStart);
+  }
+
+  @Test
+  public void shouldNotFindTaskLastUpdatedNullCreateDateBeforeQueryDate() {
+    // given
+    ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("oneTaskProcess");
+    Date afterStart = getAfterCurrentTime();
+
+    // when
+    // no update to task, lastUpdated = null
+
+    // then
+    Task taskResult = taskService.createTaskQuery().processInstanceId(processInstance.getId()).taskUpdatedAfter(afterStart).singleResult();
+    assertThat(taskResult).isNull();
+  }
+
+  @Test
+  public void shouldNotFindTaskLastUpdatedBeforeQueryDate() {
+    // given
+    ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("oneTaskProcess");
+    Task task = taskService.createTaskQuery().processInstanceId(processInstance.getId()).singleResult();
+    Date lastUpdatedBeforeUpdate = task.getLastUpdated();
+    task.setAssignee("myself");
+
+    // when
+    taskService.saveTask(task);
+
+    // then
+    assertThat(lastUpdatedBeforeUpdate).isNull();
+    Date afterUpdate = getAfterCurrentTime();
+    Task taskResult = taskService.createTaskQuery().processInstanceId(processInstance.getId()).taskUpdatedAfter(afterUpdate).singleResult();
+    assertThat(taskResult).isNull();
+    taskResult = taskService.createTaskQuery().processInstanceId(processInstance.getId()).singleResult();
+    assertThat(taskResult).isNotNull();
+    assertThat(taskResult.getLastUpdated()).isBefore(afterUpdate);
+  }
+
+  @Test
+  public void shouldNotFindTaskLastUpdatedEqualsQueryDate() {
+    // given
+    ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("oneTaskProcess");
+    Task task = taskService.createTaskQuery().processInstanceId(processInstance.getId()).singleResult();
+    task.setAssignee("myself");
+
+    // when
+    taskService.saveTask(task);
+
+    // then
+    Task taskResult = taskService.createTaskQuery().processInstanceId(processInstance.getId()).singleResult();
+    assertThat(taskResult).isNotNull();
+    assertThat(taskResult.getLastUpdated()).isNotNull();
+    taskResult = taskService.createTaskQuery().processInstanceId(processInstance.getId()).taskUpdatedAfter(taskResult.getLastUpdated()).singleResult();
+    assertThat(taskResult).isNull();
+  }
+
+  @Test
+  public void shouldNotSetLastUpdatedWithoutTaskUpdate() {
+    // given
+    ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("oneTaskProcess");
+
+    // when
+    // no update to task, lastUpdated = null
+
+    // then
+    Task taskResult = taskService.createTaskQuery().processInstanceId(processInstance.getId()).singleResult();
+    assertThat(taskResult.getLastUpdated()).isNull();
+  }
```

Comment:
Actually, this is not testing the query API but rather some base functionality of the Task. This could be moved into a general Task test class (which we already have, as far as I know). There might be other test cases like this as well here. Maybe this test class here could rather be named `TaskLastUpdatedTest`. And the specific query functionality tests could be moved into the general `TaskQueryTest` class.